src/App.tsx

```diff
@@ -7,38 +8,113 @@ import classNames from 'classnames';
 import { Loader } from './components/Loader';
+import { Post } from './types/Post';
+import { getPosts } from './api/post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [arePostLoading, setArePostLoading] = useState(false);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const loadUsers = async () => {
+    try {
+      const loadedUsers = await getUsers();
+      setUsers(loadedUsers);
+    } catch {
+      throw new Error('Failed to load users');
+    }
+  };
+
+  const loadPosts = useCallback(async () => {
+    if (selectedUser) {
+      setArePostLoading(true);
+
+      try {
+        const loadedPosts = await getPosts(selectedUser?.id);
+
+        setPosts(loadedPosts);
+        setIsLoadingError(false);
+      } catch {
+        setIsLoadingError(true);
+      } finally {
+        setArePostLoading(false);
+      }
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    loadPosts();
+
+    setSelectedPost(null);
+  }, [selectedUser, loadPosts]);
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {arePostLoading
+                  ? <Loader />
+                  : (
+                    <>
+                      {!selectedUser
+                        ? (
+                          <p data-cy="NoSelectedUser">
+                            No user selected
+                          </p>
+                        )
+                        : (
+                          <>
+                            {isLoadingError && (
+                              <div
+                                className="notification is-danger"
+                                data-cy="PostsLoadingError"
+                              >
+                                Something went wrong!
+                              </div>
+                            )}
```

Comment:
Try to avoid using nester ternary and split these conditions

Reply:
Not fixed